src/ess/reduce/nexus/_nexus_loader.py

```diff
@@ -41,10 +41,15 @@ def load_component(
         component = _unique_child_group(instrument, nx_class, group_name)
         loaded = cast(sc.DataGroup, component[selection])
         loaded['nexus_component_name'] = component.name.split('/')[-1]
-    return compute_component_position(loaded)
+    return loaded
```

Comment:
This is the heart of the change in this PR: We are no longer computing the position directly here. This is moved into an extra step.